File: Decorator/typescript/src/Condiments/Soy.ts
import { Beverage } from "../Beverages";
import CondimentDecorator from "./CondimentDecorator";

export class Soy extends CondimentDecorator {
  private _beverage: Beverage;

  constructor(beverage: Beverage) {
    super();
    this._beverage = beverage;
    this.description = `${this._beverage.description}, Soy`;
  }

  cost() {
    return 0.15 + this._beverage.cost();
  }
}

File: Decorator/c#/Condiments/Soy.cs
using Beverages;

namespace Condiments
{
  public class Soy : CondimentDecorator
  {
    private Beverage _beverage;

    public Soy(Beverage beverage)
    {
      _beverage = beverage;
      description = $"{_beverage.description}, Soy";
    }

    public override decimal cost()
    {
      return .15m + _beverage.cost();
    }
  }
}